File: backend/models/OTP.js
// backend/models/OTP.js
import mongoose from 'mongoose';

const otpSchema = new mongoose.Schema({
  donorId: { type: mongoose.Schema.Types.ObjectId, ref: 'Donor', required: true },
  code: { type: String, required: true },
  expiresAt: { type: Date, required: true },
  createdAt: { type: Date, default: Date.now, expires: 300 } // tự động xóa sau 5 phút
});

export default mongoose.model('OTP', otpSchema);

File: backend-go/internal/models/otp.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

type OTP struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
	DonorID   primitive.ObjectID `bson:"donorId" json:"donorId"`
	Code      string             `bson:"code" json:"code"`
	ExpiresAt time.Time          `bson:"expiresAt" json:"expiresAt"`
	CreatedAt time.Time          `bson:"createdAt" json:"createdAt"`
}
